src/reactviews/pages/QueryResult/queryResultPane.tsx

```diff
@@ -309,29 +261,114 @@ export const QueryResultPane = () => {
 
     const renderGridPanel = () => {
         const grids = [];
-        // execution plans only load after reading the resulting xml showplan
-        // of the query. therefore, it updates the state once the results
-        // are loaded, which causes a rendering loop if the grid
-        // gets refreshed
-        if (!metadata?.isExecutionPlan) {
-            gridRefs.current.forEach((r) => r?.refreshGrid());
-        }
 
-        let count = 0;
         for (
             let i = 0;
             i < Object.keys(metadata?.resultSetSummaries ?? []).length;
             i++
         ) {
             var batch = metadata?.resultSetSummaries[i];
             for (let j = 0; j < Object.keys(batch ?? []).length; j++) {
-                grids.push(renderGrid(i, j, count));
-                count++;
+                grids.push(renderGrid(i, j));
             }
         }
         return grids;
     };
 
+    // ========================= MESSAGE GRID =========================
```

Comment:
Can you just move this into a separate component (in a follow up PR)? Both the message and result grid? It is not very effecient to have big monolith components. Re-rendering becomes more expensive

Reply:
Yes this is what I'm planning to do, I don't want to do it in this PR so just re-ordered things and added the section comments.